content/general/concepts/abstract-factory/abstract-factory.md

```diff
@@ -0,0 +1,381 @@
+  public static final int MAX_OVERDRAFT = 500;
+
+  public BronzePersonal(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+```java
+public class CreditBuilder extends CurrentAccount {
+  // The Credit Builder does not allow for an overdraft
+  public static final int MAX_OVERDRAFT = 0;
+
+  public CreditBuilder(Customer accountHolder, int initBalance, int agreedOverdraftLimit) {
+    super(accountHolder, initBalance, agreedOverdraftLimit);
+  }
+  // Concrete overridden method
+  @Override
+  public void increaseOverdraft(int requestedLimit) {
+    if (requestedLimit < MAX_OVERDRAFT) {
+      this.setAgreedOverdraftLimit(requestedLimit);
+    }
+  }
+}
+```
+
+In each concrete current account, a `static` variable (`MAX_OVERDRAFT`) hase been provided and constructed with the correct value. This is then used in the `increaseOverdraft` method to ensure this limit is not exceeded.
+
+Below provides an abstract model `SavingsAccount` class for its concrete sub-classes to be based on:
+
+```java
+public abstract class SavingAccount {
+  private final String accountUID = UUID.randomUUID().toString();
+  private final Customer accountHolder;
+  private final LocalDate dateOpened;
+  private double balance;
+
+  public SavingAccount(Customer accountHolder, LocalDate dateOpened, double initBalance) {
+    this.accountHolder = accountHolder;
+    this.dateOpened = dateOpened;
+    this.balance = initBalance;
+  }
+
+  public void addFunds(final int amount) {
+    balance += amount;
+  }
+
+  public void withdrawFunds(final int amount) {
+    balance -= amount;
+  }
+  // Enforces concrete classes to override this method
+  public abstract void addInterest();
+
+  // Getters and setter
+}
+```
+
+Much like `CurrentAccount`, `SavingAccount` provides the common fields and constructor and requires its children to implement the `addInterest` method. This is the differentiating feature described between each savings account.
+
+Below provides the concrete implementations of `SavingAccount` for this example:
+
+```java
+public class GoldSaver extends SavingAccount {
+  // The Gold Saver has a high interest rate
+  public static final double INTEREST_RATE_MULTIPLIER = 1.05;
+
+  public GoldSaver(Customer accountHolder, LocalDate dateOpened, double initBalance) {
+    super(accountHolder, dateOpened, initBalance);
+  }
+  // Concrete overridden method
+  @Override
+  public void addInterest() {
+    if (LocalDate.now().getMonth() == this.getDateOpened().getMonth()
+            && LocalDate.now().getDayOfMonth() == this.getDateOpened().getDayOfMonth()) {
+      this.setBalance(this.getBalance() * INTEREST_RATE_MULTIPLIER);
+    }
+  }
+}
+```
+
```

Comment:
Having no text between code blocks is discouraged. Could we add in a sentence that leads in to the next code block? Something like: "This is an implementation of the Silver-level `SavingAccount`:"